packages/nx/src/plugins/js/project-graph/affected/npm-packages.ts

```diff
@@ -76,3 +85,21 @@ export const getTouchedNpmPackages: TouchedProjectLocator<
   }
   return touched;
 };
+
+function getGlobalPackages(plugins: NxJsonConfiguration['plugins']) {
+  return (plugins ?? [])
+    .map((p) => getPackageName(typeof p === 'string' ? p : p.plugin))
+    .concat('nx');
+}
+
+//# Converts import paths to package names.
+//# e.g. - `@nx/workspace`        -> `@nx/workspace`
+//#      - `@nx/workspace/plugin` -> `@nx/workspace`
+//#      - `@nx/workspace/other`  -> `@nx/workspace`
+//#      - `nx/plugin`            -> `nx`
+export function getPackageName(name: string) {
+  if (name.startsWith('@')) {
+    return name.split('/').slice(0, 2).join('/');
+  }
+  return name.split('/')[0];
+}
```

Comment:
I believe we have this here. Could you move it somewhere and export it?

https://github.com/nrwl/nx/blob/ef0dc01ffae4b079c5017b787b884aec1503a534/packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.ts#L366